Reactor Module 2025 Training material updates

aaronoaks opened the pull request:
<!--
If this PR is associated with an issue be sure to reference it
by including "refs #<issue>" or "closes #<issue>" (e.g., #closes #1234).

If this PR implements an enhancement that does not have an existing issue, please add the following:
-->

## Reason
Reactor Module training will be held in May 2025, material needs to be updated with latest features and best practices.

## Design
Material is part of the Tutorial 4: Meshing

## Impact
Updates to Meshing Tutorial content.


<!--
If this PR implements a bug fix, please create an issue for the bug and reference the issue.
-->

shikhar413 committed:
Update reactor module trainings for 2025 training.

Closes #30071

moosebuild commented on large_media:
```diff
@@ -1 +1 @@
-Subproject commit 7df25da9663f229a3a89db8e8e3bfe66caf9f712
+Subproject commit 08a1bc4838fcbcb0e2b36f601e62fb1ac0b09850
```
**Caution!** This contains a submodule update

moosebuild replied:
**Caution!** This contains a submodule update

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/2742760/), step Generate coverage on 047efd7 wanted to post the following:


## Framework coverage

<table>
  <thead>
    <tr>
      <th></th>
      <th>8ca331</th>
      <th colspan=4>#30071 047efd</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>85.29%</td>
      <td>85.29%</td>
      <td>+0.00%</td>
      <td>-</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>109035</td>
      <td>109036</td>
      <td>+1</td>
      <td>0</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>18808</td>
      <td>18807</td>
      <td>-1</td>
      <td>0</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/30071/coverage_diff/framework)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/30071/coverage/framework)


## Modules coverage

Coverage did not change

## Full coverage reports

<details>
<summary>Reports</summary>

- [`framework`](https://mooseframework.inl.gov/docs/PRs/30071/coverage/framework)
- [`chemical_reactions`](https://mooseframework.inl.gov/docs/PRs/30071/coverage/chemical_reactions)
- [`combined`](https://mooseframework.inl.gov/docs/PRs/30071/coverage/combined)
- [`contact`](https://mooseframework.inl.gov/docs/PRs/30071/coverage/contact)
- [`electromagnetics`](https://mooseframework.inl.gov/docs/PRs/30071/coverage/electromagnetics)
- [`external_petsc_solver`](https://mooseframework.inl.gov/docs/PRs/30071/coverage/external_petsc_solver)
- [`fluid_properties`](https://mooseframework.inl.gov/docs/PRs/30071/coverage/fluid_properties)
- [`fsi`](https://mooseframework.inl.gov/docs/PRs/30071/coverage/fsi)
- [`functional_expansion_tools`](https://mooseframework.inl.gov/docs/PRs/30071/coverage/functional_expansion_tools)
- [`geochemistry`](https://mooseframework.inl.gov/docs/PRs/30071/coverage/geochemistry)
- [`heat_transfer`](https://mooseframework.inl.gov/docs/PRs/30071/coverage/heat_transfer)
- [`level_set`](https://mooseframework.inl.gov/docs/PRs/30071/coverage/level_set)
- [`misc`](https://mooseframework.inl.gov/docs/PRs/30071/coverage/misc)
- [`navier_stokes`](https://mooseframework.inl.gov/docs/PRs/30071/coverage/navier_stokes)
- [`optimization`](https://mooseframework.inl.gov/docs/PRs/30071/coverage/optimization)
- [`peridynamics`](https://mooseframework.inl.gov/docs/PRs/30071/coverage/peridynamics)
- [`phase_field`](https://mooseframework.inl.gov/docs/PRs/30071/coverage/phase_field)
- [`porous_flow`](https://mooseframework.inl.gov/docs/PRs/30071/coverage/porous_flow)
- [`ray_tracing`](https://mooseframework.inl.gov/docs/PRs/30071/coverage/ray_tracing)
- [`rdg`](https://mooseframework.inl.gov/docs/PRs/30071/coverage/rdg)
- [`reactor`](https://mooseframework.inl.gov/docs/PRs/30071/coverage/reactor)
- [`richards`](https://mooseframework.inl.gov/docs/PRs/30071/coverage/richards)
- [`scalar_transport`](https://mooseframework.inl.gov/docs/PRs/30071/coverage/scalar_transport)
- [`solid_mechanics`](https://mooseframework.inl.gov/docs/PRs/30071/coverage/solid_mechanics)
- [`solid_properties`](https://mooseframework.inl.gov/docs/PRs/30071/coverage/solid_properties)
- [`stochastic_tools`](https://mooseframework.inl.gov/docs/PRs/30071/coverage/stochastic_tools)
- [`thermal_hydraulics`](https://mooseframework.inl.gov/docs/PRs/30071/coverage/thermal_hydraulics)
- [`xfem`](https://mooseframework.inl.gov/docs/PRs/30071/coverage/xfem)

</details>



This comment will be updated on new commits.

Commit:
Removed old RGMB step12 example documentation.

shikhar413 committed:
Update tutorial04_meshing website pages

moosebuild commented on large_media:
```diff
@@ -1 +1 @@
-Subproject commit 1191583486c0aa2ecfb9b276dfd50fe942ecf258
+Subproject commit 853a643eb4ac24a151fcdaa6d77bff76ca5d4b6a
```
**Caution!** This contains a submodule update

shikhar413 commented:
@eshemon in case you want to review what the reactor module tutorial documentation looks like with Yinbin and my changes incorporated, you can take a look at these temporary links:

- https://mooseframework.inl.gov/docs/PRs/30071/site/getting_started/examples_and_tutorials/tutorial04_meshing/index.html
- https://mooseframework.inl.gov/docs/PRs/30071/site/getting_started/examples_and_tutorials/tutorial04_meshing/presentation/index.html#/

eshemon commented:
[like]  Shemon, Emily R. reacted to your message:
________________________________
From: shikhar413 ***@***.***>
Sent: Thursday, March 13, 2025 7:24:06 PM
To: idaholab/moose ***@***.***>
Cc: Shemon, Emily R. ***@***.***>; Mention ***@***.***>
Subject: Re: [idaholab/moose] Reactor Module 2025 Training material updates (PR #30071)

@ eshemon in case you want to review what the reactor module tutorial documentation looks like with Yinbin and my changes incorporated, you can take a look at these temporary links: https: //mooseframework. inl. gov/docs/PRs/30071/site/getting_started/examples_and_tutorials/tutorial04_meshing/index. html
ZjQcmQRYFpfptBannerStart
This Message Is From an External Sender
This message came from outside your organization.

ZjQcmQRYFpfptBannerEnd

@eshemon<https://urldefense.us/v3/__https://github.com/eshemon__;!!G_uCfscf7eWS!Z4sasRcHmh2w8x82AMW_uKHExlz6cyVHt4Qw_xlnbpA3xRrAF7dDgO7JzMjRmLNapN8hkTZn_iVtcNBxOAhtTMk$> in case you want to review what the reactor module tutorial documentation looks like with Yinbin and my changes incorporated, you can take a look at these temporary links:

  *   https://mooseframework.inl.gov/docs/PRs/30071/site/getting_started/examples_and_tutorials/tutorial04_meshing/index.html<https://urldefense.us/v3/__https://mooseframework.inl.gov/docs/PRs/30071/site/getting_started/examples_and_tutorials/tutorial04_meshing/index.html__;!!G_uCfscf7eWS!Z4sasRcHmh2w8x82AMW_uKHExlz6cyVHt4Qw_xlnbpA3xRrAF7dDgO7JzMjRmLNapN8hkTZn_iVtcNBxv60b8Eo$>
  *   https://mooseframework.inl.gov/docs/PRs/30071/site/getting_started/examples_and_tutorials/tutorial04_meshing/presentation/index.html#/<https://urldefense.us/v3/__https://mooseframework.inl.gov/docs/PRs/30071/site/getting_started/examples_and_tutorials/tutorial04_meshing/presentation/index.html*/__;Iw!!G_uCfscf7eWS!Z4sasRcHmh2w8x82AMW_uKHExlz6cyVHt4Qw_xlnbpA3xRrAF7dDgO7JzMjRmLNapN8hkTZn_iVtcNBxR6jvoE8$>

—
Reply to this email directly, view it on GitHub<https://urldefense.us/v3/__https://github.com/idaholab/moose/pull/30071*issuecomment-2722468996__;Iw!!G_uCfscf7eWS!Z4sasRcHmh2w8x82AMW_uKHExlz6cyVHt4Qw_xlnbpA3xRrAF7dDgO7JzMjRmLNapN8hkTZn_iVtcNBxpgaNebk$>, or unsubscribe<https://urldefense.us/v3/__https://github.com/notifications/unsubscribe-auth/AT6JR7JK4CT7AMHOUUFZLVL2UHLNNAVCNFSM6AAAAABYXF52UGVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDOMRSGQ3DQOJZGY__;!!G_uCfscf7eWS!Z4sasRcHmh2w8x82AMW_uKHExlz6cyVHt4Qw_xlnbpA3xRrAF7dDgO7JzMjRmLNapN8hkTZn_iVtcNBxoTbo7sA$>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

[shikhar413]shikhar413 left a comment (idaholab/moose#30071)<https://urldefense.us/v3/__https://github.com/idaholab/moose/pull/30071*issuecomment-2722468996__;Iw!!G_uCfscf7eWS!Z4sasRcHmh2w8x82AMW_uKHExlz6cyVHt4Qw_xlnbpA3xRrAF7dDgO7JzMjRmLNapN8hkTZn_iVtcNBxpgaNebk$>

@eshemon<https://urldefense.us/v3/__https://github.com/eshemon__;!!G_uCfscf7eWS!Z4sasRcHmh2w8x82AMW_uKHExlz6cyVHt4Qw_xlnbpA3xRrAF7dDgO7JzMjRmLNapN8hkTZn_iVtcNBxOAhtTMk$> in case you want to review what the reactor module tutorial documentation looks like with Yinbin and my changes incorporated, you can take a look at these temporary links:

  *   https://mooseframework.inl.gov/docs/PRs/30071/site/getting_started/examples_and_tutorials/tutorial04_meshing/index.html<https://urldefense.us/v3/__https://mooseframework.inl.gov/docs/PRs/30071/site/getting_started/examples_and_tutorials/tutorial04_meshing/index.html__;!!G_uCfscf7eWS!Z4sasRcHmh2w8x82AMW_uKHExlz6cyVHt4Qw_xlnbpA3xRrAF7dDgO7JzMjRmLNapN8hkTZn_iVtcNBxv60b8Eo$>
  *   https://mooseframework.inl.gov/docs/PRs/30071/site/getting_started/examples_and_tutorials/tutorial04_meshing/presentation/index.html#/<https://urldefense.us/v3/__https://mooseframework.inl.gov/docs/PRs/30071/site/getting_started/examples_and_tutorials/tutorial04_meshing/presentation/index.html*/__;Iw!!G_uCfscf7eWS!Z4sasRcHmh2w8x82AMW_uKHExlz6cyVHt4Qw_xlnbpA3xRrAF7dDgO7JzMjRmLNapN8hkTZn_iVtcNBxR6jvoE8$>

—
Reply to this email directly, view it on GitHub<https://urldefense.us/v3/__https://github.com/idaholab/moose/pull/30071*issuecomment-2722468996__;Iw!!G_uCfscf7eWS!Z4sasRcHmh2w8x82AMW_uKHExlz6cyVHt4Qw_xlnbpA3xRrAF7dDgO7JzMjRmLNapN8hkTZn_iVtcNBxpgaNebk$>, or unsubscribe<https://urldefense.us/v3/__https://github.com/notifications/unsubscribe-auth/AT6JR7JK4CT7AMHOUUFZLVL2UHLNNAVCNFSM6AAAAABYXF52UGVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDOMRSGQ3DQOJZGY__;!!G_uCfscf7eWS!Z4sasRcHmh2w8x82AMW_uKHExlz6cyVHt4Qw_xlnbpA3xRrAF7dDgO7JzMjRmLNapN8hkTZn_iVtcNBxoTbo7sA$>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

loganharbour commented:
> [like] Shemon, Emily R. reacted to your message:

I think this is the first time I've seen an outlook reaction end up in a github comment 😆

eshemon commented:
@loganharbour  I "liked it" on Outlook, then I thought to myself, "that probably didn't do anything, I'll log on to Github", then I saw it WORKED!

aaronoaks commented:
@loganharbour I noticed when looking through the slides that our old slides don't lay out the way I remember. The code blocks look to be larger font, and take up more space than before. E.g., in this slide below, the code block and figure used to be two 50/50 side by side, now the code block considerably overpowers the figure. If the presentation visual style has been updated since the last training, do you know if there are directives we area able to apply to revert the behavior for our presentation?

https://mooseframework.inl.gov/docs/PRs/30071/site/getting_started/examples_and_tutorials/tutorial04_meshing/presentation/index.html#/12/3

aaronoaks commented:
@loganharbour We also found while testing out of examples files on INL HPC that some of our test cases were producing input errors. We have corrections for these input files, but it doesn't look like CIVET noticed the issues. Are these test cases being exercised by CIVET, or is there something that needs to be updated to check them?

https://github.com/aaronoaks/moose/tree/reactor-training-2025/tutorials/tutorial04_meshing/app/test/tests/

GiudGiud commented on modules/doc/content/getting_started/examples_and_tutorials/tutorial04_meshing/step11_rgmb_het_hom.md:
```diff
@@ -0,0 +1,128 @@
+# Reactor Geometry Mesh Builder Example: Conversion of Heterogeneous to Homogeneous Sodium-Cooled Fast Reactor Core Mesh (ABTR)
+
```
maybe add a note

this step is Griffin only, the regular MOOSE tutorial ends here

shikhar413 replied:
Most of this tutorial is MOOSE only (everything up to het mesh generation), only the last step (conversion from het to hom) is Griffin only

GiudGiud replied:
I agree but we should signal the end of the MOOSE-only content smh

shikhar413 replied:
I will add a comment about this

shikhar413 replied:
Added a comment to this file where the endpoint is

GiudGiud commented on modules/doc/content/getting_started/examples_and_tutorials/tutorial04_meshing/step11_rgmb_het_hom.md:
```diff
@@ -0,0 +1,128 @@
+- Use a unique [PinMeshGenerator.md] block for each pin with a unique geometrical configuration and/or region ID composition
+- region_ids is a 2-dimensional array containing region IDs (essentially materials). The first row of the array represents the 2D radial regions (from center of the pin to outermost region) for the bottom layer of the pin. Each subsequent row assigns IDs on another axial level, from bottom to top. In this case, each pin has 3 radial regions (fuel, clad, background), this array is a column pertain ...
+- While the mesh is still 2D during this step, the axially dependent region IDs are stored for later use during the extrusion step.
+
```
```suggestion
!alert-end!

```

GiudGiud commented on modules/doc/content/getting_started/examples_and_tutorials/tutorial04_meshing/step11_rgmb_het_hom.md:
```diff
@@ -0,0 +1,128 @@
+         caption=ABTR RGMB heterogeneous assembly example.
+         block=Mesh/fuel_assembly_1
+
+Tips
```
same format could look beter

GiudGiud commented on modules/doc/content/getting_started/examples_and_tutorials/tutorial04_meshing/step11_rgmb_het_hom.md:
```diff
@@ -0,0 +1,128 @@
+Tips
+
+- Use a unique [AssemblyMeshGenerator.md] block for each assembly with a unique geometrical configuration, region ID composition, and/or inventory of pin structures
+- background_region_ids is a 1-dimensional array containing region IDs for each axial layer of the background region. duct_region_ids is a 2-dimensional array containing regions IDs for the duct region. The first row of the array represents the 2D duct regions (from innermost to outermost duct region) for the bottom layer of the assembly. Each subsequent row assigns IDs on another axial level, fr ...
```
```suggestion
- [!param](/Mesh/AssemblyMeshGenerator/background_region_ids) is a 1-dimensional array containing region IDs for each axial layer of the background region. duct_region_ids is a 2-dimensional array containing regions IDs for the duct region. The first row of the array represents the 2D duct regions (from innermost to outermost duct region) for the bottom layer of the assembly. Each subsequent row assigns IDs on another axial level, from bottom to top.
```
same for any other param please. it prevents deprecation of the training content

shikhar413 replied:
Updated, fixed other instances too. Let me know if I missed anything

GiudGiud commented on modules/doc/content/getting_started/examples_and_tutorials/tutorial04_meshing/step11_rgmb_het_hom.md:
```diff
@@ -0,0 +1,128 @@
+In order to define homogeneous assembly structures to stitch into the core, we use [PinMeshGenerator.md] once again ([AssemblyMeshGenerator.md] is only used for structures that consist of lattices of pins).
+
+- +To define single assemblies directly with PinMeshGenerators for stitching with [CoreMeshGenerator.md]+, [PinMeshGenerator.md] is used with [!param](/Mesh/PinMeshGenerator/use_as_assembly) set to `true`.
+- In addition, [!param](/Mesh/PinMeshGenerator/homogenized) = `true` is used to indicate that this region is homogenized and [SimpleHexagonGenerator.md] should be called to discretize the assembly instead of [PolygonConcentricCircleMeshGenerator.md].
```
the SimpleHexagonGenerator does not actually show up in the input ?

this could confuse users. I would not mention it

shikhar413 replied:
Done

GiudGiud commented on modules/doc/content/getting_started/examples_and_tutorials/tutorial04_meshing/step11_rgmb_het_hom.md:
```diff
@@ -0,0 +1,128 @@
+         caption=ABTR RGMB homogeneous assembly example.
+         block=Mesh/reflector_assembly
+
+Tips
```
same comments

GiudGiud commented on modules/doc/content/getting_started/examples_and_tutorials/tutorial04_meshing/step11_rgmb_het_hom.md:
```diff
@@ -0,0 +1,128 @@
+
+Now that all heterogeneous and homogeneous assemblies have been defined, they are placed into a lattice using [CoreMeshGenerator.md]. While [CoreMeshGenerator.md] still requires a perfect hexagonal pattern like [PatternedHexMeshGenerator.md], it automatically handles dummy assembly creation and deletion. The user need only provide a fake mesh input reference `dummy` (this object has not been actu ...
+
+- Since we want to extrude the 2D core, we use the [!param](/Mesh/CoreMeshGenerator/extrude)=`true` parameter within [CoreMeshGenerator.md]. The step simultaneously extrudes the geometry and applies the regions IDs to all axial layers as defined in the [PinMeshGenerator.md] objects.
```
it also processes the assembly MG ducts and background axial parameters too?

shikhar413 replied:
Updated to include references to both PMG and AMG objects

GiudGiud commented on modules/doc/content/getting_started/examples_and_tutorials/tutorial04_meshing/step11_rgmb_het_hom.md:
```diff
@@ -0,0 +1,128 @@
+Now that all heterogeneous and homogeneous assemblies have been defined, they are placed into a lattice using [CoreMeshGenerator.md]. While [CoreMeshGenerator.md] still requires a perfect hexagonal pattern like [PatternedHexMeshGenerator.md], it automatically handles dummy assembly creation and deletion. The user need only provide a fake mesh input reference `dummy` (this object has not been actu ...
+
+- Since we want to extrude the 2D core, we use the [!param](/Mesh/CoreMeshGenerator/extrude)=`true` parameter within [CoreMeshGenerator.md]. The step simultaneously extrudes the geometry and applies the regions IDs to all axial layers as defined in the [PinMeshGenerator.md] objects.
+- Behind the scenes, extra element IDs `assembly_id` and `plane_id` are automatically applied.
```
```suggestion
- Behind the scenes, extra element IDs `assembly_id` and `plane_id` are automatically added to the relevant elements
```

GiudGiud commented on modules/doc/content/getting_started/examples_and_tutorials/tutorial04_meshing/step11_rgmb_het_hom.md:
```diff
@@ -0,0 +1,128 @@
+       caption=RGMB-generated 3D heterogeneous core mesh, colored by various IDs defined on the mesh.
+       style=width:75%;display:block;margin-left:auto;margin-right:auto;
+
+## Equivalent homogeneous core using Griffin's EquivalentCoreMeshGenerator
```
or should the tutorial end right here for MOOSE-only?

GiudGiud replied:
(the griffin content can stay for convenience)

shikhar413 replied:
Added a description here to indicate end of MOOSE content and beginning of Griffin content

GiudGiud commented on tutorials/tutorial04_meshing/app/test/tests/reactor_examples/rgmb_empire/rgmb_empire.i:
where is Empire being used? i n which step?

shikhar413 replied:
I haven't gotten around to finishing the documentation for this example yet. I plan to add this to a follow-up PR, unless you want me to push the documentation updates to this PR itself?

GiudGiud replied:
either work for me

shikhar413 replied:
Will work on this this week. Feel free to approve this PR if the documentation updates haven't been included before all comments are addressed

shikhar413 replied:
@aaronoaks this can be resolved now

GiudGiud commented on tutorials/tutorial04_meshing/doc/content/getting_started/examples_and_tutorials/tutorial04_meshing/presentation/tutorial04_step11_rgmb_het_hom.md:
```diff
@@ -0,0 +1,123 @@
+## Conversion of Heterogeneous to Homogeneous Sodium-Cooled Fast Reactor Core Mesh (ABTR)
+
+This example illustrates the use of RGMB mesh generators to define a pin-resolved heterogeneous 3D hexagonal geometry core (ABTR ([!cite](shemon2015abtr)), and using Griffin to convert this mesh to an equivalent homogenized core mesh with automatic geometry construction and region ID assignment. The final mesh constructed will be similar to the one presented earlier in this tutorial using base me ...
+
```
xplicitly call out how far a non-griffin user can go ?

shikhar413 replied:
Done

GiudGiud commented on tutorials/tutorial04_meshing/doc/content/getting_started/examples_and_tutorials/tutorial04_meshing/presentation/tutorial04_step11_rgmb_het_hom.md:
```diff
@@ -0,0 +1,123 @@
+In order to define homogeneous assembly structures to stitch into the core, we use [PinMeshGenerator.md] once again ([AssemblyMeshGenerator.md] is only used for structures that consist of lattices of pins).
+
+- +To define single assemblies directly with PinMeshGenerators for stitching with [CoreMeshGenerator.md]+, [PinMeshGenerator.md] is used with [!param](/Mesh/PinMeshGenerator/use_as_assembly) set to `true`.
+- In addition, [!param](/Mesh/PinMeshGenerator/homogenized) = `true` is used to indicate that this region is homogenized and [SimpleHexagonGenerator.md] should be called to discretize the assembly instead of [PolygonConcentricCircleMeshGenerator.md].
```
same commment

GiudGiud commented on tutorials/tutorial04_meshing/doc/content/getting_started/examples_and_tutorials/tutorial04_meshing/presentation/tutorial04_step13_advanced_tools.md:
```diff
@@ -2,6 +2,19 @@
+ 
+ The quadratic element options can be selected by setting `tri_element_type` as TRI6 or TRI7, and `quad_element_type` as QUAD8 or QUAD9 in the 2D mesh generators.
+ 
+ The 2D-to-3D mesh generators such as [AdvancedExtruderGenerator.md] detects 2D quadratic elements and create 3D quadratic elements accordingly.
```
```suggestion
 The 2D-to-3D mesh generators such as [AdvancedExtruderGenerator.md] detect 2D quadratic elements and create 3D quadratic elements accordingly.
```

GiudGiud commented on tutorials/tutorial04_meshing/doc/content/getting_started/examples_and_tutorials/tutorial04_meshing/presentation/tutorial04_step13_advanced_tools.md:
```diff
@@ -77,3 +90,41 @@ This [FillBetweenPointVectorsGenerator.md] object generates a transition layer b
+!---
+
+## RevolveGenerator
+
```
add a one line description ?

GiudGiud reviewed (changes requested):
updates looks great to me!

shikhar413 commented on tutorials/tutorial04_meshing/doc/content/getting_started/examples_and_tutorials/tutorial04_meshing/presentation/tutorial04_step07_extra_element_ids.md:
```diff
@@ -63,6 +64,8 @@
 
 - Pin and Assembly IDs are applied during creations of assemblies and core, respectively.
 
+- [FlexiblePatternGenerator.md] also supports creation of reporting IDs for each unique pin structure (See "Advanced Meshing Tools" section)
```
@GiudGiud do you know of any way to reference this section in the slides? I tried
```
See ["Advanced Meshing Tools"](tutorial04_meshing/presentation/index.md#/15) section
```
and that works, where 15 is the slide number of the section, but this will point to the wrong section if the tutorial content is reordered

GiudGiud replied:
see shortcut links here
https://mooseframework.inl.gov/python/MooseDocs/extensions/core.html

that should work by adding an id to a section

shikhar413 replied:
Not having luck with this when it comes to the presentation. I added id=sec_name in the header I want to link to and then [#sec_name] in this section, but the link does not work

shikhar413 replied:
Keeping as `See ["Advanced Meshing Tools"](tutorial04_meshing/presentation/index.md#/15) section` for now. Do you want me to add an issue for this? Referencing other sections with `id= ` doesn't seem to work for presentations

GiudGiud replied:
ah ok. Yeah let's have an issue

shikhar413 replied:
https://github.com/idaholab/moose/issues/30297

shikhar413 commented on tutorials/tutorial04_meshing/doc/content/getting_started/examples_and_tutorials/tutorial04_meshing/presentation/tutorial04_step08_abtr.md:
```diff
@@ -241,6 +241,8 @@ We briefly touch on mesh generation for the Coarse Mesh Finite Difference (CMFD)
 
 This ABTR example uses option 1, where the CMFD acceleration uses the same mesh as the fine mesh, so no additional mesh generation is performed.
 
+So far, the example shown in this section has relied on explicit region ID mappings for the homogeneous core. Section "RGMB Example: Heterogeneous to Homogeneous Conversion for Fast Reactor Core" will explore how the Reactor Geometry Mesh Builder can be used to generate the heterogeneous input mesh and automatically define the homogeneous core mesh from the heterogeneous mesh specifications.
```
Tried this and got a 404 not found on that link

shikhar413 replied:
Setting this to `Section ["RGMB Example: Heterogeneous to Homogeneous Conversion for Fast Reactor Core"](tutorial04_meshing/presentation/index.md#/11)` for now until a better alternative works for presentations

shikhar413 committed:
Make updates to RGMB content based on comments from GiudGiud

shikhar413 commented on modules/doc/content/getting_started/examples_and_tutorials/tutorial04_meshing/step11_rgmb_het_hom.md:
```diff
@@ -0,0 +1,128 @@
+         block=Mesh/fuel_pin_1
+
+Tips
+
```
Added this to all Tips in this file

GiudGiud commented:
Please avoid putting handles in commit numbers. I would get an email every time someone rebases

GiudGiud commented:
yes please

Commit:
Updates to address review comments.

Commit:
Updates to tutorial test cases to correct errors from changing input syntax.

aaronoaks commented:
@GiudGiud thank you for the review! I believe we have addressed most comments, but I left two open for potential follow-ups. I also had addressed a couple questions to Logan above but maybe he is occupied with other things. Are you able to answer these?

shikhar413 committed:
Add documentation for RGMB EMPIRE example

shikhar413 commented:
Since this PR was still pending approval I pushed my updates to add the documentation for the RGMB Empire portion of the tutorial. Only the last commit needs to be reviewed, assuming no other comments before the latest commit

moosebuild commented on large_media:
```diff
@@ -1 +1 @@
-Subproject commit a796dadb5d7bf81e3c4b5826e8beabf613e71a52
+Subproject commit fdf470ed8e7ee0cd5b65a624e2e136ed1f26c05f
```
**Caution!** This contains a submodule update

moosebuild replied:
**Caution!** This contains a submodule update

GiudGiud commented on tutorials/tutorial04_meshing/doc/content/getting_started/examples_and_tutorials/tutorial04_meshing/presentation/tutorial04_step11_rgmb_het_hom.md:
```diff
@@ -0,0 +1,124 @@
+
+- Since [!param](/Mesh/ReactorMeshParams/flexible_assembly_stitching) = `true` in [ReactorMeshParams.md], the different assembly types are stitched together without hanging nodes.
+- Dummy assemblies do not need to be defined explicitly and instead can be referenced in the core lattice input by specifying a name in [!param](/Mesh/CoreMeshGenerator/dummy_assembly_name).
+- This is the last step you can run if you do not have Griffin exectuable access.
```
```suggestion
- This is the last step you can run if you do not have Griffin executable access.
```

GiudGiud commented on tutorials/tutorial04_meshing/doc/content/getting_started/examples_and_tutorials/tutorial04_meshing/presentation/tutorial04_step12_rgmb_empire.md:
```diff
@@ -0,0 +1,117 @@
+
+## ReactorMeshParams
+
+[ReactorMeshParams.md] contains global mesh/geometry parameters including whether the final mesh is 2D or 3D, Cartesian or hexagonal, assembly pitch, and the axial discretization for the final extruded geometry. Here we also invoke the option to enable flexible assembly stitching with [!param](/Mesh/ReactorMeshParams/flexible_assembly_stitching), so that dissimilar assembly structutes can be stit ...
```
```suggestion
[ReactorMeshParams.md] contains global mesh/geometry parameters including whether the final mesh is 2D or 3D, Cartesian or hexagonal, assembly pitch, and the axial discretization for the final extruded geometry. Here we also invoke the option to enable flexible assembly stitching with [!param](/Mesh/ReactorMeshParams/flexible_assembly_stitching), so that dissimilar assembly structures can be stitched into the reactor core without any hanging nodes. This parameter needs to be set to true in order to use [ControlDrumMeshGenerator.md].
```

GiudGiud commented on modules/doc/content/getting_started/examples_and_tutorials/tutorial04_meshing/step12_rgmb_empire.md:
```diff
@@ -0,0 +1,129 @@
+
+## ReactorMeshParams
+
+[ReactorMeshParams.md] contains global mesh/geometry parameters including whether the final mesh is 2D or 3D, Cartesian or hexagonal, assembly pitch, and the axial discretization for the final extruded geometry. This information will be accessible to the other RGMB mesh generators and consistently used. Here we also invoke the option to enable flexible assembly stitching with [!param](/Mesh/React ...
```
```suggestion
[ReactorMeshParams.md] contains global mesh/geometry parameters including whether the final mesh is 2D or 3D, Cartesian or hexagonal, assembly pitch, and the axial discretization for the final extruded geometry. This information will be accessible to the other RGMB mesh generators and consistently used. Here we also invoke the option to enable flexible assembly stitching with [!param](/Mesh/ReactorMeshParams/flexible_assembly_stitching), so that dissimilar assembly structures can be stitched into the reactor core without any hanging nodes. This parameter needs to be set to true in order to use [ControlDrumMeshGenerator.md].
```

GiudGiud reviewed (commented):
a few typos but otherwise good to go imo

shikhar413 committed:
Fix typos in training documentation

shikhar413 commented on modules/doc/content/getting_started/examples_and_tutorials/tutorial04_meshing/step11_rgmb_het_hom.md:
```diff
@@ -0,0 +1,140 @@
+
+## Heterogeneous core using CoreMeshGenerator
+
+At this point, we have defined a 3D heterogeneous ABTR Now that all heterogeneous and homogeneous assemblies have been defined, they are placed into a lattice using [CoreMeshGenerator.md]. While [CoreMeshGenerator.md] still requires a perfect hexagonal pattern like [PatternedHexMeshGenerator.md], it automatically handles dummy assembly creation and deletion. The user need only provide a fake mesh ...
```
Actually removing this line because a full 3-D mesh hasn't been generated by this point yet

GiudGiud commented:
@kyriv1980 what is the ETA for https://github.com/idaholab/moose/pull/30229 ?
If not soon we need to revert your large_media changes

kyriv1980 commented:
> @kyriv1980 what is the ETA for #30229 ? If not soon we need to revert your large_media changes

It's gonna be soon. Early next week?

aaronoaks commented:
@GiudGiud any updates on this one? The training is the coming up (May 5-6), so we want to be sure this gets merged in and on the live site in time.

GiudGiud commented:
I'm watching the PR that needs to go in first but yes we won't miss the deadline

shikhar413 commented:
Rebasing with next since #30229 was merged

moosebuild commented on large_media:
```diff
@@ -1 +1 @@
-Subproject commit b93604476d0b5b0d1913e22e123395dee572e8cb
+Subproject commit fdf470ed8e7ee0cd5b65a624e2e136ed1f26c05f
```
**Caution!** This contains a submodule update

GiudGiud commented:
Thanks! Good to go

eshemon commented:
[like]  Shemon, Emily R. reacted to your message:
________________________________
From: Guillaume Giudicelli ***@***.***>
Sent: Thursday, April 24, 2025 10:59:53 AM
To: idaholab/moose ***@***.***>
Cc: Shemon, Emily R. ***@***.***>; Mention ***@***.***>
Subject: Re: [idaholab/moose] Reactor Module 2025 Training material updates (PR #30071)

Merged #30071 into next. — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you were mentioned. Message ID: <idaholab/moose/pull/30071/issue_event/17388150866@ github. com> ‍ ‍ ‍ ‍ ‍ ‍ ‍ ‍ ‍ ‍ ‍
ZjQcmQRYFpfptBannerStart
This Message Is From an External Sender
This message came from outside your organization.

ZjQcmQRYFpfptBannerEnd

Merged #30071<https://urldefense.us/v3/__https://github.com/idaholab/moose/pull/30071__;!!G_uCfscf7eWS!aHpZVP6PH0iT38zsoLuTifu1z2h3YayF8YS_iR-tX0oCcn6UEunPSWKSVsMNIa0AodOKB6DbHWFqSBIH9hmEaFI$> into next.

—
Reply to this email directly, view it on GitHub<https://urldefense.us/v3/__https://github.com/idaholab/moose/pull/30071*event-17388150866__;Iw!!G_uCfscf7eWS!aHpZVP6PH0iT38zsoLuTifu1z2h3YayF8YS_iR-tX0oCcn6UEunPSWKSVsMNIa0AodOKB6DbHWFqSBIHoJGMlAM$>, or unsubscribe<https://urldefense.us/v3/__https://github.com/notifications/unsubscribe-auth/AT6JR7KBDJIEBJ6XN7P2VPT23C72TAVCNFSM6AAAAABYXF52UGVHI2DSMVQWIX3LMV45UABCJFZXG5LFIV3GK3TUJZXXI2LGNFRWC5DJN5XDWMJXGM4DQMJVGA4DMNQ__;!!G_uCfscf7eWS!aHpZVP6PH0iT38zsoLuTifu1z2h3YayF8YS_iR-tX0oCcn6UEunPSWKSVsMNIa0AodOKB6DbHWFqSBIHaAz7X3c$>.
You are receiving this because you were mentioned.Message ID: ***@***.***>
